CMakeLists.txt

```diff
@@ -19,8 +22,7 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
-find_package(MPI REQUIRED)
-find_package(Threads REQUIRED)
```

Comment:
I agree, one would not expect that `find_package(Threads REQUIRED)` is required. Though, for some combination of C++ compiler and MPI implementation this was necessary in the past. We need to double check, if we can actually omit this.